### ISSUE TITLE
update google style coverage guide with note that checkstyle can not cover "No static import for classes" chapter

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#4005]
Based on the discussion at https://github.com/checkstyle/checkstyle/issues/3755#issuecomment-285851574 and according to Google Java Style Guide (3 November 2016 version) [3.3.4 No static import for classes](https://github.com/google/styleguide/commit/db0a26320f3e930c6ea7225ed53539b4fb31310c#diff-b6c9191ee5ccb7a5ef170e98e51d94d8R210) to cover the rule from the guide AvoidStaticImport should have a new option which will restrict the scope of validation only to static imports of nested classes. We should take into account that the guide says that the rule is applicable only to static **nested** classes. 

**Should be taken into account during the implementation**
To determine that there is an import of static nested class we can analyze import statement to find out whether it includes supposed nested class name.

For example, say we have the class MyClass which contains static nested class MyStaticNestedClass:

```java
package nested;

public class MyClass {
    public static class MyStaticNestedClass {
        public static void foo() {
        }
    }
```

Lets assume, that class MyTestClass imports static nested class MyStaticNestedClass:

```java
package main;

import static nested.MyClass.MyStaticNestedClass; // violation

public class Main {
    public static void main(String[] args) {
        MyStaticNestedClass.foo();
    }
}
```

We can check the import statement and extract the information about nested class. However, not only static nested classes can be imported this way, but also interfaces and enums. It means that we cannot say for sure that there is an import of static nested class. 

**Option name and configuration**
Option name: checkOnlyNestedClasses.
Default value: false (to avoid breaking of backward compatibility).

**Comment by MEZk** [comment]
@romani @rnveach 
If you have any ideas on how to solve the problem with importing of enums and interfaces, please share.

**Comment by rnveach** [comment]
> import static nested.MyClass.MyStaticNestedClass; // violation

@MEZk If I am reading this correctly, this looks like something we can't support because it requires knowing the makeup of an external class.

We can identify a static import, but we can't tell if we are importing a class, method, or field. For all we know `MyStaticNestedClass` could be any of those things from just looking at `Main`. It is only when we look at `MyClass` do we know what it is.

**Comment by romani** [comment]
We have no information about classes out of single file.
By name we can not certainly tell is class static or not.

In scope of this issues we need to upgrade coverage report to mention that we can not cover this point.

**Comment by romani** [comment]
issue title has changed, to describe what is required to do.

**Comment by romani** [comment]
update to coverage is provided at https://github.com/checkstyle/checkstyle/commit/ef1efd0e0b851bc9369858ce5d1d93af7b0fc8fa - we can not cover this point.
